script/instances-0.6.0-to-0.7.0.sh

```diff
@@ -7,25 +7,40 @@ fi
 
 set -euo pipefail
 
-instanceDir="$1"
-backupDir="$instanceDir.bak_$(date +%s)"
+jq_version="$(jq --version 2>&1)" # redirecting stderr https://github.com/stedolan/jq/issues/1452
+continue=0
+if [[ $jq_version != *"1.5"* ]]; then
+  echo "WARNING: This script was tested against jq-1.5 and should be run against that as well."
+  read -p "WARNING: Looks like you are using $jq_version. Continue anyway? [y/n] " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    continue=1
+  fi
+else
+  continue=1
+fi
 
-echo "Backing up $instanceDir to $backupDir."
-cp -r $instanceDir $backupDir
-echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
-for instanceFile in $instanceDir/*.json; do
-  echo "- Converting $instanceFile"
-  tmpInstanceFile="$instanceFile.tmp"
-  cat $instanceFile | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' > $tmpInstanceFile
-  mv $tmpInstanceFile $instanceFile
-done
+if [[ "$continue" == 1 ]]; then
+  instanceDir="$1"
+  backupDir="$instanceDir.bak_$(date +%s)"
 
-echo "Conversion finished. Looks like everything went well."
-read -p "Delete $backupDir? [y/n] " -n 1 -r
-echo
-if [[ $REPLY =~ ^[Yy]$ ]]; then
-  echo "Deleting $backupDir."
-  rm -rf $backupDir
-else
-  echo "Keeping $backupDir for now. You have to delete it manually."
+  echo "Backing up $instanceDir to $backupDir."
+  cp -r $instanceDir $backupDir
+  echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
+  for instanceFile in $instanceDir/*.json; do
+    echo "- Converting $instanceFile"
+    tmpInstanceFile="$instanceFile.tmp"
+    cat $instanceFile | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' > $tmpInstanceFile
```

Comment:
Running cat in front of a pipe is redundant; use stdin redirection instead, ie, replace `cat $instanceFile | jq …` with `jq … < $instanceFile`.  

Furthermore, please quote all occurrences of variables in this line, ie, `$tmpInstanceFile` and $instanceFile.

Last but not least, use a proper temporary file with `mktemp` to avoid accidentally or incidentally overwriting files through a predictable file name.